app/models/spree/variant_decorator.rb

```diff
@@ -84,6 +84,54 @@ def self.indexed
     ]
   end
 
+  # Overriding `Spree::Variant.on_hand` in old Spree versions.
+  # It doesn't exist in newer Spree versions.
+  def on_hand
+    if respond_to? :total_on_hand
+      # This is Spree 2.0
+      total_on_hand
+    elsif Spree::Config[:track_inventory_levels] && !on_demand
+      count_on_hand
+    else
+      Float::INFINITY
+    end
+  end
+
+  # Overriding `Spree::Variant.on_hand=` in old Spree versions.
+  # It doesn't exist in newer Spree versions.
+  def on_hand=(new_level)
+    error = 'Cannot set on_hand value when Spree::Config[:track_inventory_levels] is false'
+    raise error unless Spree::Config[:track_inventory_levels]
+
+    self.count_on_hand = new_level unless on_demand
+  end
+
+  # Overriding Spree::Variant.count_on_hand in old Spree versions.
+  # It doesn't exist in newer Spree versions.
+  def count_on_hand
+    if respond_to? :total_on_hand
+      # This is Spree 2.0
+      total_on_hand
+    else
+      # We assume old Spree and call ActiveRecord's method.
+      # We can't call the same method on Variant, because we are overwriting it.
+      super
+    end
+  end
+
+  # Overriding `Spree::Variant.count_on_hand=` in old Spree versions.
+  # It doesn't exist in newer Spree versions.
+  def count_on_hand=(new_level)
```

Comment:
Are these methods necessary? `Spree::Variant` doesn't have `count_on_hand` or `count_on_hand=`methods, it uses the `on_hand` methods as getters and setters.

Reply:
Or is this related to Variant Overrides?

Reply:
In Spree 1.3 the `variants` table has a `count_on_hand` column. So ActiveRecord adds these methods automatically and we have a few calls to these methods in our code base. In order to keep the old API alive, we need these methods.